python/private/pypi/pip_compile.bzl

```
@@ -147,6 +157,18 @@ def pip_compile(
     # cheap way to detect the bazel version
     _bazel_version_4_or_greater = "propeller_optimize" in dir(native)
 
+    # setuptools (the default python build tool) attempts to find user
+    # configuration in the user's home direcotory. This seems to work fine on
+    # linux and macOS, but fails on Windows, so we conditionally provide a fake
+    # USERPROFILE env variable to allow setuptools to proceed without finding
+    # user-provided configuration.
+    env = kwargs.get("env", {})
+    if "USERPROFILE" not in env:
+        kwargs["env"] = env | select({
+            "@@platforms//os:windows": {"USERPROFILE": "Z:\\FakeSetuptoolsHomeDirectoryHack"},
+            "//conditions:default": {},
+        })
```

Pointed out by a coworker of mine (not @ lpulley this time!), `not in` will fail if `env` is a `select`.

We came up with the following to fix this, but the `dict` union operator (`|`) isn't supported pre bazel 6 and we weren't sure which approach was preferred.

```suggestion
        kwargs["env"] = select({
        "@@platforms//os:windows": {"USERPROFILE": "Z:\\FakeSetuptoolsHomeDirectoryHack"},
        "//conditions:default": {},
    }) | kwargs.get("env", {})
```

---

We only need to support bazel 6 and above with the latest version of `rules_python` so this is fine.